ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -692,3 +736,131 @@ func validateTaskNetworkConfig(taskNetworkConfig *state.TaskNetworkConfig) error
 
 	return nil
 }
+
+// startNetworkPacketLossFault invokes the linux TC utility tool to start the network-packet-loss fault.
+func (h *FaultHandler) startNetworkPacketLossFault(taskMetadata *state.TaskResponse, request types.NetworkPacketLossRequest) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	interfaceName := taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].NetworkInterfaces[0].DeviceName
+	networkMode := taskMetadata.TaskNetworkConfig.NetworkMode
+	// If task's network mode is awsvpc, we need to run nsenter to access the task's network namespace.
+	nsenterPrefix := ""
+	if networkMode == ecs.NetworkModeAwsvpc {
+		nsenterPrefix = fmt.Sprintf(nsenterCommandString, taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].Path)
+	}
+	lossPercent := *request.LossPercent
+
+	// Command to be executed:
+	// <nsenterPrefix> tc qdisc add dev <interfaceName> root handle 1: prio priomap 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
+	// <nsenterPrefix> "tc qdisc add dev <interfaceName> parent 1:1 handle 10: netem loss <lossPercentage>%"
+	tcAddQdiscRootCommandComposed := nsenterPrefix + fmt.Sprintf(tcAddQdiscRootCommandString, interfaceName)
+	cmdOutput, err := h.runExecCommand(ctx, tcAddQdiscRootCommandComposed)
+	if err != nil {
+		return err
```

Comment:
nit: could we add a log statement here in case an error occurred when running the command?

Reply:
Yes. Will add logging for all exec failures